### PR TITLE
Add xyz.eclipium.TetraSwap

### PR DESCRIPTION
**Pull request by SpeedCode210**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: xyz.eclipium.TetraSwap.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            "name": "TetraSwap",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir /app/bin && cp -R TetraSwap/* /app/bin/ && mkdir /app/share && mkdir /app/share/applications && cp TetraSwap.desktop /app/share/applications/ && mkdir /app/share/icons && mkdir /app/share/icons/hicolor && mkdir /app/share/icons/hicolor/512x512 && mkdir /app/share/icons/hicolor/128x128 && mkdir /app/share/icons/hicolor/64x64 && mkdir /app/share/icons/hicolor/128x128/apps && mkdir /app/share/icons/hicolor/512x512/apps && mkdir /app/share/icons/hicolor/64x64/apps && cp TetraSwap.png /app/share/icons/hicolor/512x512/apps/ && cp 128.png /app/share/icons/hicolor/128x128/apps/TetraSwap.png && cp 64.png /app/share/icons/hicolor/64x64/apps/TetraSwap.png && mkdir /app/share/metainfo && cp TetraSwap.appdata.xml /app/share/metainfo/ && bzip2 -d /app/bin/TetraSwap_Data/sharedassets0.assets.resS.bz2"
+            ],
+            "sources": [
+                {
+                    "type": "dir",
+                    "path": "Files"
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:28]: you need to tell were to download it.

File: xyz.eclipium.TetraSwap.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--share=network"
+  	],
+    "modules": [
+        {
+            "name": "TetraSwap",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir /app/bin && cp -R TetraSwap/* /app/bin/ && mkdir /app/share && mkdir /app/share/applications && cp TetraSwap.desktop /app/share/applications/ && mkdir /app/share/icons && mkdir /app/share/icons/hicolor && mkdir /app/share/icons/hicolor/512x512 && mkdir /app/share/icons/hicolor/128x128 && mkdir /app/share/icons/hicolor/64x64 && mkdir /app/share/icons/hicolor/128x128/apps && mkdir /app/share/icons/hicolor/512x512/apps && mkdir /app/share/icons/hicolor/64x64/apps && cp TetraSwap.png /app/share/icons/hicolor/512x512/apps/ && cp 128.png /app/share/icons/hicolor/128x128/apps/TetraSwap.png && cp 64.png /app/share/icons/hicolor/64x64/apps/TetraSwap.png && mkdir /app/share/metainfo && cp TetraSwap.appdata.xml /app/share/metainfo/ && bzip2 -d /app/bin/TetraSwap_Data/sharedassets0.assets.resS.bz2"
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:23]: you should split this into multiple lines / commands for readability

File: TetraSwap.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ <metadata_license>CC0-1.0</metadata_license>
+ <project_license>LicenseRef-proprietary</project_license>
+ <name>TetraSwap</name>
+ <icon type="remote" width="1024" height="1024">https://launcher.eclipium.xyz/Images/TetraSwap.png</icon>
+ <summary lang="en_US">A simple but challenging futuristic puzzle game!</summary>
+ <summary lang="fr_FR">Un jeu de réflexion futuriste simple mais difficile!</summary>
+ <categories>
+    <category>Game</category>
+ </categories>
+  <description lang="en_US">
```
Comment (hfiguiere) [inline TetraSwap.appdata.xml:14]:
```suggestion
  <description>
```

File: TetraSwap.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,82 @@
+<!-- Copyright 2021 Eclipium -->
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+ <id>xyz.eclipium.TetraSwap</id>
+ <metadata_license>CC0-1.0</metadata_license>
+ <project_license>LicenseRef-proprietary</project_license>
+ <name>TetraSwap</name>
+ <icon type="remote" width="1024" height="1024">https://launcher.eclipium.xyz/Images/TetraSwap.png</icon>
+ <summary lang="en_US">A simple but challenging futuristic puzzle game!</summary>
```
Comment (hfiguiere) [inline TetraSwap.appdata.xml:9]:
```suggestion
 <summary>A simple but challenging futuristic puzzle game!</summary>
```

File: TetraSwap.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    <content_attribute id="social-info">mild</content_attribute>
+  </content_rating>
+  <recommends>
+  <memory>2048</memory> <!-- recommend at least 2GiB of memory -->
+</recommends>
+<supports>
+  <control>pointing</control>
+  <control>keyboard</control>
+  <control>touch</control>
+</supports>
```
Comment (hfiguiere) [inline TetraSwap.appdata.xml:81]: You need a `<releases>` tag

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by SpeedCode210** [comment]
In this commit, I made all the necessary changes

**Comment by hfiguiere** [comment]
bot, build xyz.eclipium.TetraSwap

**Comment by hfiguiere** [comment]
since this copies x86_64 binaries, you also need to to this:
https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

To only build x86_64.

File: xyz.eclipium.TetraSwap.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    "runtime-version": "21.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "TetraSwap",
+	"rename-appdata-file": "TetraSwap.appdata.xml",
+	"rename-icon": "TetraSwap",
+	"rename-desktop-file": "TetraSwap.desktop",
+    "finish-args": [
+    "--share=ipc",
+    "--socket=wayland",
+    "--socket=fallback-x11",
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:13]: is wayland actually supported?

File: xyz.eclipium.TetraSwap.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+				"mkdir /app/share/icons/hicolor/128x128",
+				"mkdir /app/share/icons/hicolor/64x64",
+				"mkdir /app/share/icons/hicolor/128x128/apps",
+				"mkdir /app/share/icons/hicolor/512x512/apps",
+				"mkdir /app/share/icons/hicolor/64x64/apps",
+				"cp TetraSwap.png /app/share/icons/hicolor/512x512/apps/",
+				"cp 128.png /app/share/icons/hicolor/128x128/apps/TetraSwap.png",
+				"cp 64.png /app/share/icons/hicolor/64x64/apps/TetraSwap.png",
+				"mkdir /app/share/metainfo",
+				"cp TetraSwap.appdata.xml /app/share/metainfo/"
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:43]:
```suggestion
				"install TetraSwap.appdata.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml"
```

And remove the `rename-appdata-file`

**Comment by SpeedCode210** [comment]
I solved the problem here

File: xyz.eclipium.TetraSwap.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--share=network"
+  	],
+    "modules": [
+        {
+            "name": "TetraSwap",
+            "buildsystem": "simple",
+            "build-commands": [
+            	"ls",
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:23]:
```suggestion
```

File: xyz.eclipium.TetraSwap.json [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            	"mkdir /app/bin",
+				"cp TetraSwap /app/bin/",
+				"cp -r TetraSwap_Data /app/bin/",
+				"cp UnityPlayer.so /app/bin/",
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:27]:
```suggestion
				"install -Dm755 -t /app/bin/ TetraSwap",
				"install -t /app/bin UnityPlayer.so",
				"cp -r TetraSwap_Data /app/bin/",
```

File: xyz.eclipium.TetraSwap.json [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+				"mkdir /app/share",
+				"mkdir /app/share/applications",
+				"cp TetraSwap.desktop /app/share/applications/",
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:30]:
```suggestion
				"install -Dm644 TetraSwap.desktop /app/share/applications/${FLATPAK_ID}.desktop",
```

and remove the `rename-desktop-file` at the top

File: xyz.eclipium.TetraSwap.json [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+				"mkdir /app/share/icons/hicolor",
+				"mkdir /app/share/icons/hicolor/512x512",
+				"mkdir /app/share/icons/hicolor/128x128",
+				"mkdir /app/share/icons/hicolor/64x64",
+				"mkdir /app/share/icons/hicolor/128x128/apps",
+				"mkdir /app/share/icons/hicolor/512x512/apps",
+				"mkdir /app/share/icons/hicolor/64x64/apps",
+				"cp TetraSwap.png /app/share/icons/hicolor/512x512/apps/",
+				"cp 128.png /app/share/icons/hicolor/128x128/apps/TetraSwap.png",
+				"cp 64.png /app/share/icons/hicolor/64x64/apps/TetraSwap.png",
```
Comment (hfiguiere) [inline xyz.eclipium.TetraSwap.json:41]:
```suggestion
				"install -Dm644 TetraSwap.png /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.json",
				"install -Dm644 128.png /app/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png",
				"install -Dm644 64.png /app/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png",
```

and remove the `rename-icon` at the top.

File: TetraSwap.desktop [hunk 12]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Type=Application
+Version=1.0
+Name=TetraSwap
+Comment=A puzzle game made by Eclipium !
+Icon=TetraSwap
```
Comment (hfiguiere) [inline TetraSwap.desktop:6]:
```suggestion
Icon=xyz.eclipium.TetraSwap
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: xyz.eclipium.TetraSwap.json (hunk 6 above)
Comment (SpeedCode210) [reply xyz.eclipium.TetraSwap.json:13]: Yes

**Comment by SpeedCode210** [comment]
All corrections have been made

**Comment by hfiguiere** [comment]
bot, build xyz.eclipium.TetraSwap

**Comment by SpeedCode210** [comment]
bot, build xyz.eclipium.TetraSwap

File: TetraSwap.appdata.xml [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+ </description>
+   <description lang="fr_FR">
+  <p>
+TetraSwap est un jeu de réflexion futuriste disponible sur téléphone tablette et ordinateur.
+Le jeu est composé d'un plateau composé de plusieurs cases colorées (2 à 4 couleurs) ainsi que de jetons colorés et d'un jeton gris, le but du jeu étant de placer tous les jetons sur leurs cases de couleur correspondante, cela pourrait vous sembler facile mais détrompez vous ! Car le seul moyen de les déplacer est d'utiliser le jeton gris, compliqué à maîtriser.
+Il y a 3 modes de jeu :
+Classique : Possédant 48 niveaux (peut augmenter au fil des mises à jour), avec diverses mécaniques de jeu introduites au fil des niveaux.
+Aléatoire : Le jeu génère aléatoirement un plateau avec un nombre de cases et de couleurs définis.
+Journalier : Relevez les défis journaliers et essayez de dépasser vos amis dans le classement.
+ </p>
```
Comment (hfiguiere) [inline TetraSwap.appdata.xml:32]:
```suggestion
  <p>
TetraSwap est un jeu de réflexion futuriste disponible sur téléphone tablette et ordinateur.
</p>
<p>
Le jeu est composé d'un plateau composé de plusieurs cases colorées (2 à 4 couleurs) ainsi que de jetons colorés et d'un jeton gris, le but du jeu étant de placer tous les jetons sur leurs cases de couleur correspondante, cela pourrait vous sembler facile mais détrompez vous ! Car le seul moyen de les déplacer est d'utiliser le jeton gris, compliqué à maîtriser.
</p>
<p>
Il y a 3 modes de jeu :
</p>
<ul>
<li>Classique : Possédant 48 niveaux (peut augmenter au fil des mises à jour), avec diverses mécaniques de jeu introduites au fil des niveaux.</li>
<li>Aléatoire : Le jeu génère aléatoirement un plateau avec un nombre de cases et de couleurs définis.</li>
<li>Journalier : Relevez les défis journaliers et essayez de dépasser vos amis dans le classement.</li>
 </ul>
```

And likely same for the english version.
The problem is the version long single `<p>` that end up being displayed no the way you want.

Comment (SpeedCode210) [reply TetraSwap.appdata.xml:32]: I corrected the problem with another way, thanks

**Comment by SpeedCode210** [comment]
Hello, is everything okay for adding the app to flathub ?

**Pull request closed without merging**